platform/app/prisma/migrations/20260804120000_dataset_record_project_dataset_index/migration.sql

```
@@ -0,0 +1,24 @@
+-- touching the heap: 3.5ms for a typical project, 165ms for the largest,
+-- on a faithful 2.58M-row reproduction of the production distribution.
+-- It also serves the project-scoped record reads on the dataset editor page.
+--
+-- LOCKING NOTE: plain `CREATE INDEX` takes a SHARE lock on "DatasetRecord" —
+-- concurrent reads keep working, writes (uploads, appends, row edits) block
+-- until the build finishes. The build reads the heap once: ~2.8s on a 2.58M-row
+-- reproduction, so expect single-digit seconds against production. Dataset
+-- writes are low-frequency and this runs during deploy, so the brief write
+-- pause is acceptable. `CREATE INDEX CONCURRENTLY` would avoid it entirely but
+-- cannot run inside a transaction, which the Prisma migration setup requires.
+CREATE INDEX "DatasetRecord_projectId_datasetId_idx" ON "DatasetRecord"("projectId", "datasetId");
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)